Ocbc daya

daya448 opened the pull request:
Modified tracks as per the Design Doc. 
Tested Cohere Only.
Testing Pending on openai and msmarco tracks.

daya448 committed:
custom challenges as per Design Doc

daya448 committed:
2 sub benchmarks added to all 3 data sets

daya448 committed:
minor fixes

daya448 committed:
Added new track

Base template added for new track.

daya448 committed:
new track changes : in memory vs ondisk

daya448 committed:
fixes after testing cohere track

mbrunnert commented on msmarco-v2-vector-memory/README.md:
Readme file should be updated to match the track

mbrunnert commented on msmarco-v2-vector-memory/challenges/run-benchmark.json:
We might need to create individual operations / corpora for each individual step. Otherwise I think that the data load will only load the initial 1M documents in each step? Ideally it should continue to the next 1M documents in the data set, to avoid duplicate vectors in the index.

mbrunnert commented on msmarco-v2-vector-memory/operations/default.json:
```diff
@@ -0,0 +1,76 @@
+  "recency": {{recency | default(0)}},
+},
+{
+  "name": "parallel-documents-indexing",
```
We should clean up unused operations etc.

mbrunnert commented on msmarco-v2-vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,56 @@
+      "name": "index-benchmark",
+      "operation": "index-benchmark",
+      "warmup-time-period": {{ initial_indexing_bulk_warmup | default(40) | int }},
+      "clients": {{ initial_indexing_bulk_indexing_clients | default(1) | int }}
```
Search clients may need to be adjusted upwards to reach desired qps

mbrunnert reviewed (approved):
Looks good overall, lets test and adjust based on our findings. A few fairly minor comments.

elangoshifts commented on openai_vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,67 @@
+    "description": "Challenge to close, restore, and benchmark index with specified operations.",
+    "schedule": [
+      {
+        "name": "close-benchmark-index",
```
Once the index is closed, the wait-until merge finishes doesn't complete, since there are no indicies being returned.

elangoshifts commented on openai_vector/challenges/initial-load.json:
```diff
@@ -0,0 +1,75 @@
+        "name": "set-refresh-rate",
+        "operation": {
+          "operation-type": "raw-request",
+          "path": "vdb910-openai/_settings",
```
Path has to start with /

elangoshifts commented on openai_vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,67 @@
+        }
+      },
+      {
+        "name": "index-benchmark",
```
since the snapshot already has the initial - index data , do we need to perform this again ?

elangoshifts commented on openai_vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,67 @@
+        "operation": "check-cluster-health"
+      },
+      {
+        "name": "wait-until-merges-finish-after-index",
```
Since we have closed the index and the only indices we have access to is vdb910-openai, the GET request returns below response
`{"_shards":{"total":0,"successful":0,"failed":0},"_all":{"primaries":{},"total":{}},"indices":{}}`
this doesn't match the condition

elangoshifts commented on openai_vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,67 @@
+        "operation": "knn-search-10-100",
+        "warmup-iterations": 100,
+        "target-throughput": 150,
+        "iterations": 10000,
```
this line can be removed

mbrunnert committed:
minor fixes to allow the msmarco track to run

mbrunnert committed:
Changed snapshot api calls to use single repository

mbrunnert committed:
Changed repo name to match onsite snapshot repo

mbrunnert committed:
fixed issue with msmarco corpora

mbrunnert committed:
initial index performance tuning

mbrunnert committed:
initial index performance tuning

mbrunnert committed:
fixed cluster health request

mbrunnert committed:
fixed cluster health request

mbrunnert committed:
adjusted number of documents indexed to meet reqs and reduce running time of benchmarks

mbrunnert committed:
further tuning of parameters. added explicit id for relevance benchmarking

mbrunnert committed:
fix pipeline reference

mbrunnert committed:
fix pipeline reference. changed order of search and index benchmarks

mbrunnert committed:
fix pipeline reference. changed order of search and index benchmarks

elangoshifts commented on msmarco-v2-vector/operations/default.json:
```diff
@@ -16,7 +16,28 @@
+  "operation-type": "bulk",
+  "corpora": "msmarco-v2-initial-indexing",
+  "bulk-size": {{bulk_size | default(500)}},
+  "ingest-percentage": {{ingest_percentage | default(10)}},
```
have to add time-period of 30 mins

mbrunnert committed:
cleaning up benchmarks for runs next week

elangoshifts committed:
wait for active shards only a number or all

A number controlling to how many active shards to wait for, all to wait for all shards in the cluster to be active, or 0 to not wait. Defaults to 0.

elangoshifts committed:
Remove test data generation dependencies

These dependencies are only required to generate test data file (embeddings) from raw data. This step has already been performed and hence removing

elangoshifts committed:
Use _base_url variable for location of test data

base url can then be overridden using track-params when running in airgap env. Default goes to rally-tracks.elastic.co

elangoshifts committed:
refer to check-cluster-health defined in operations

update-index still fails due to missing documents

elangoshifts commented on msmarco-v2-vector/challenges/run-benchmark.json:
```diff
@@ -0,0 +1,64 @@
+    },
+    {
+      "name": "index-benchmark",
+      "operation": "index-benchmark",
```
should we use parallel-documents-indexing (which uses a different corpora), because this operation ends in about a minute and there is no change in doc count size. May be it is picking the same docs and sending.

mbrunnert committed:
finalise first v1 of msmarco

mbrunnert committed:
adjust clients

mbrunnert committed:
adjust ingest percentage

mbrunnert committed:
fixed update challenge

mbrunnert committed:
tuned update clients

mbrunnert committed:
adjusted speed of update benchmark

mbrunnert committed:
adjustments to meet latency requirements

mbrunnert committed:
fixed index benchmark time period

mbrunnert committed:
tuning cohere and openai benchmarks

mbrunnert committed:
update fix

mbrunnert committed:
adjusting index clients

mbrunnert committed:
increase search warm up time

mbrunnert committed:
start work on memory

mbrunnert committed:
json bugs

mbrunnert committed:
memory benchmark bugs

mbrunnert committed:
fix memory checksums

mbrunnert committed:
updated ingestion to use appropriate number of docs

mbrunnert committed:
added tools to generate corpora

mbrunnert committed:
fixes in scripts to generate data

mbrunnert committed:
fixes in scripts to generate data

mbrunnert committed:
corpora sorting

mbrunnert committed:
removed compression from memory corpora

mbrunnert committed:
work on splitting corpus

mbrunnert committed:
refactor memory track to reflect new corpi

mbrunnert committed:
fixed main challenge

mbrunnert committed:
separated memory into initial load and run

mbrunnert committed:
separated memory into initial load and run

mbrunnert committed:
separated memory into initial load and run

mbrunnert committed:
Added hybrid search

mbrunnert committed:
tuned memory benchmark

elangoshifts committed:
Throttle down inserts and search closer to test plan

mbrunnert committed:
tuned memory benchmark

elangoshifts committed:
Throttle down inserts to test plan levels

elangoshifts committed:
Throttle down create index operation as per test plan

elangoshifts committed:
Throttle down update throughput

mbrunnert committed:
tuned memory benchmark

mbrunnert committed:
Increased search times to 10m

elangoshifts committed:
Use openai index

mbrunnert committed:
delete benchmark

mbrunnert committed:
bugfix

mbrunnert committed:
bugfix

mbrunnert committed:
Added esrally track for loading relevance index

mbrunnert committed:
Add source exclude

mbrunnert committed:
tune indexing speed

mbrunnert committed:
additional docs challenge

mbrunnert committed:
additional docs challenge

mbrunnert committed:
additional docs challenge

mbrunnert committed:
add msmarco-passage-ranking

mbrunnert committed:
added english analysed field for hybrid search

mbrunnert committed:
added track to load trec-covid data

mbrunnert committed:
bug fix trec
